serialization/serialization.go

```diff
@@ -1,205 +0,0 @@
-package serialization
-
-import (
-	"errors"
-
-	bls12381 "github.com/consensys/gnark-crypto/ecc/bls12-381"
-	"github.com/consensys/gnark-crypto/ecc/bls12-381/fr"
-	"github.com/crate-crypto/go-proto-danksharding-crypto/internal/kzg"
-	"github.com/crate-crypto/go-proto-danksharding-crypto/internal/utils"
-)
-
-// This is the number of 32 byte slices a blob can contain.
-// We use the nomenclature `FIELD_ELEMENTS_PER_BLOB` because
-// each field element when serialized is 32 bytes
-//
-// These 32 byte slices may not be _valid_, to which an error
-// will be returned on deserialization.
-//
-// This constant is set at the 4844 protocol level and is not
-// related to any cryptographic assumptions.
-const ScalarsPerBlob = 4096
-
-// This is the number of bytes needed to represent a
-// group element in G1 when compressed.
-const CompressedG1Size = 48
-
-// This is the number of bytes needed to represent a
-// group element in G2 when compressed.
-const CompressedG2Size = 96
-
-// This is the number of bytes needed to represent a field
-// element corresponding to the order of the G1 group.
-const SerializedScalarSize = 32
-
-type Scalar = [SerializedScalarSize]byte
-type G1Point = [CompressedG1Size]byte
-type G2Point = [CompressedG2Size]byte
-
-// A blob is a flattened representation for a serialized polynomial
-type Blob = [ScalarsPerBlob * SerializedScalarSize]byte
-
-// This is a misnomer, its KZGWitness
-type KZGProof = G1Point
-type KZGCommitment = G1Point
-
-type Commitment = G1Point
-type Commitments = []Commitment
-
-func SerializeG1Point(affine bls12381.G1Affine) G1Point {
-	return affine.Bytes()
-}
-
-// This method will return an error if the point is not on the group
-// or if the point is not in the correct subgroup.
-//
-// [validate_kzg_g1](https://github.com/ethereum/consensus-specs/blob/3a2304981a3b820a22b518fe4859f4bba0ebc83b/specs/deneb/polynomial-commitments.md#validate_kzg_g1)
-// [bytes_to_kzg_commitment](https://github.com/ethereum/consensus-specs/blob/3a2304981a3b820a22b518fe4859f4bba0ebc83b/specs/deneb/polynomial-commitments.md#bytes_to_kzg_commitment)
-// [bytes_to_kzg_proof](https://github.com/ethereum/consensus-specs/blob/3a2304981a3b820a22b518fe4859f4bba0ebc83b/specs/deneb/polynomial-commitments.md#bytes_to_kzg_proof)
-func DeserializeG1Point(serPoint G1Point) (bls12381.G1Affine, error) {
-	var point bls12381.G1Affine
-
-	_, err := point.SetBytes(serPoint[:])
```

Comment:
You are relying on gnark for uncompressing and subgroup checking. That makes sense. In the same manner, c-kzg relies on blst for the same operation. I wonder if these two libraries have identical behavior. 

They probably do because they both implement the IETF spec, but I wonder if there are ways an attacker can make one library accept something and the other to reject.